### ISSUE TITLE
Out of stock products available to buy within category view

### ISSUE THREAD

**Issue by cb2004** [cubecart/v6#1103]
This code does not work within content.category.php

```
{if $product.available <= 0}
<div class="row collapse">
    <div class="small-12 columns">
        <input type="submit" value="{$LANG.common.unavailable}" class="button small disabled expand marg-top" disabled>
    </div>
</div>
{elseif $product.ctrl_stock && !$CATALOGUE_MODE}
<div class="row collapse">
    <div class="small-4 columns">
        <input type="text" name="quantity" value="1" class="quantity text-center">
    </div>
    <div class="small-8 columns">
        <button type="submit" value="{$LANG.catalogue.add_to_basket}" class="button small postfix">{$LANG.catalogue.add_to_basket}</button>
    </div>
</div>
{elseif !$CATALOGUE_MODE}
<div class="row collapse">
    <div class="small-12 columns">
        <input type="submit" value="{$LANG.catalogue.out_of_stock_short}" disabled class="button disabled expand small">
    </div>
</div>
{/if}
```

{$product.available} always returns 1. A quick {$product|print_r} shows that:

`{if $product.available <= 0}`

Works with:

`{if $product.stock_level <= 0}`

I am not sure if there is a bug with {$product.available} or if this is a mistake in the template.

**Comment by Noodleyman** [comment]
product.available is for use with coupons. 

'available' => ($coupon['allowed_uses'] > 0) ? $coupon['allowed_uses']-$coupon['count'] : 0,

**Comment by cb2004** [comment]
The title of the bug was wrong. Basically you have the option to add to basket an out-of-stock product from the category view.

**Comment by Noodleyman** [comment]
under what stock conditions can you purchase an out of stock item?

**Comment by cb2004** [comment]
It doesn't actually add to basket, once it goes through to the product you get the error message that the product is out of stock. But you shouldnt be given the option to add to basket at all.

**Comment by Noodleyman** [comment]
ok, so the issue is that the product is out of stock, but you still see the option to add to basket and you expect that option to be removed?

**Comment by cb2004** [comment]
That is correct yes.

**Comment by Noodleyman** [comment]
ok, so that is currently expected behaviour, so I don't think it's a bug, it just doesn't work how you may want it to.

If a product is out of stock, what would be your ideal way to handle the condition?

**Comment by cb2004** [comment]
Ideally just an out of stock message. The user doesnt need to attempt to add to basket, to then be taken to the product page to be told the product is out of stock, then they have to go back to the category page they were on.

**Comment by Noodleyman** [comment]
you can always remove the add to basket button if the stock level is enabled and less than 0?

**Comment by jasehead** [comment]
Any fix also needs to be tested with a product matrix and that there are four occurrences in three files affected: Foundation/templates: content.category.php, content.homepage.php and content.product.php

**Comment by jasehead** [comment]
First attempt for content.category.php (not tested for .homepage or .product or with product matrix)
Line 111 - change this
{elseif $product.ctrl_stock && !$CATALOGUE_MODE}
to this
{elseif $product.ctrl_stock && $product.stock_level > 0 && !$CATALOGUE_MODE}

**Comment by jasehead** [comment]
In content.product.php at line 114 there is
{if ($CTRL_ALLOW_PURCHASE) && (!$CATALOGUE_MODE)}

So that may be another way to test if an Add to Basket button should be available or not.

**Comment by jasehead** [comment]
In content.product.php at line 114 there is
{if ($CTRL_ALLOW_PURCHASE) && (!$CATALOGUE_MODE)}

So that may be another way to test if an Add to Basket button should be available or not (because you may still want to allow purchases where stock levels are zero or less).

**Comment by jasehead** [comment]
UPDATE: I think my above attempt at a fix might break the Add to Basket button on the product page at least - currently trying to track down why my buttons are showing Unavailable even when items are in stock and marked as available. So, nice try but I don't think it worked.

**Comment by jasehead** [comment]
Tested and working.  This fixes the issue with the Add to Basket button showing if stock level is less than zero.

skins/foundation/templates/content.homepage.php
skins/foundation/templates/content.category.php

Find this (more than one occurrence):
{elseif $product.ctrl_stock && !$CATALOGUE_MODE}

Change to:
{elseif $product.ctrl_stock && $product.stock_level > 0 && !$CATALOGUE_MODE}

**Comment by Noodleyman** [comment]
I ran into this problem today, and finally better understand what you were saying and the issue described.

@abrookbanks , there is a bug here which I've tested in CC612, stock foundation skin.
If you view a product which has stock control enabled and is out of stock, then it shows the out of stock messages.

If you view the same product in the category view, it shows as in stock and available for purchase.

I resolved it by adding an additional check. If stock level is enabled AND stock level greater than 0 AND catalogue mode is not enabled. I had to convert the stock level from a string to an INT so the greater than 0 would work correctly.

`{elseif $product.ctrl_stock && $product.stock_level|intval > 0 && !$CATALOGUE_MODE}`
